tests/tst_decode.f

```diff
@@ -59,39 +71,39 @@ program tst_decode
      &  0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1,
      &  0, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1,
      &  1, 1, 1, 0, 1, 0, 0, 1, 1, 1, 1 /)
-      
-!    /* clang-format on */
-!        gribfield* gfld = NULL;
-
-       if (ret .eq. g2_info(cgrib, listsec0, listsec1, &numfields, &numlocal)) != 0)
-     &   return G2_ERROR
+       
+ 
+        call gb_info(fgrib, fgrib_len, listsec0, listsec1,
+     &  numfields, numlocal, maxlocal, iret)
+       if (iret .ne. 0)  print *, G2_ERROR
 
        do i =0,2
-         if (listsec0(i) .ne. listsec0_ok(i)) return G2_ERROR
-         i=i+1
+         if (listsec0(i) .ne. listsec0_ok(i)) print *, G2_ERROR
+!         i=i+1
        end do
 
        do i =0,12
-         if (listsec1(i) .ne. listsec1_ok(i)) return G2_ERROR
-         i=i+1
+         if (listsec1(i) .ne. listsec1_ok(i)) print *, G2_ERROR
+!         i=i+1
        end do
 
-       if (numfields .ne. 1) return G2_ERROR
-       if (numlocal .ne. 0)  return G2_ERROR
+       if (numfields .ne. 1) print *, G2_ERROR
+       if (numlocal .ne. 0)  print *, G2_ERROR
 
-       if ((ret .eq. g2_getfld(cgrib, 1, 1, 1, &gfld)) .ne. 0) return G2_ERROR
+       call gf_getfld(fgrib, fgrib_len, 1, 1, 1, gfld, iret)
+       if (iret .ne. 0)  print *, G2_ERROR
 
-       if (gfld%version .ne. 2) return G2_ERROR
+       if (gfld%version .ne. 2) print *, G2_ERROR
 
-       if (gfld%ndpts .ne. 121) return G2_ERROR    !/* 11x11 grid */
+       if (gfld%ndpts .ne. 121) print *, G2_ERROR    !/* 11x11 grid */
 
        do i =0,(gfld%ndpts-1) 
-        if (gfld%fld(i) .ne. fld_ok(i)) return G2_ERROR
-         i=i+1
+        if (gfld%fld(i) .ne. fld_ok(i)) print *, G2_ERROR
```

Comment:
Seems like you are just printing the error message, and still returning 0 from the program.

When there is an error the fortran code must call STOP 2 (or any number other than 0).

That will cause the program to exit with a non-zero result, and CMake will then know that the test failed.

Reply:
agree